src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -66,18 +74,28 @@ def insert_curation(sorting_key: dict, parent_curation_id: str = 'original',
             metrics = {}
 
         # generate a unique ID
+        num = (Curation & sorting_key).fetch('curation_num')
+        if len(num) > 0:
+            num.sort()
+            curation_num = num[-1] + 1
+        else:
+            curation_num = 0
+
+        # generate a unique ID string
         found = True
         while found:
-            id = 'C_' + str(uuid.uuid4())[:8]
-            if len((Curation & {'curation_id': id}).fetch()) == 0:
+            curation_id = 'C_' + str(uuid.uuid4())[:8]
+            if len((Curation & {'curation_id': curation_id}).fetch()) == 0:
```

Comment:
1. is it necessary to go through the entire table to generate a unique ID? currently the curation id the first 8 hexadecimal digits of a random UUID. since 16^8 is greater than 4 billion, it is highly unlikely that we will get the same id twice here. If you're really worried, then you can use the first 12 digits instead of 8. 
2. currently the primary key is defined as a composite of the foreign key from `SpikeSorting` and `curation_id`. this means `curation_id` just needs to be unique up to a sorting. so it is perfectly fine for two entries to have the same curation id if they come from different sortings. put another way, it is even less likely for the few curations associated with a sorting to have the same id. 
3. it _is_ plausible that we may have few hundred entries in this table in the future. in that case, comparing the curation id against every entry in the table would be consume time/memory unnecessarily.
4. if we really want to give this kind of unique ID to a sorting, then it alone would suffice to uniquely identify an entry. in that case, it should be the sole primary key (the foreign key from `SpikeSorting` should be moved to non-primary key). I don't really think this is good because then the user has to rely on a random ID to identify a curation. Plus conceptually the sorting should be used to identify it.
5. and this is why I advocate the curation id to be an int that is unique up to a sorting. these issues have been discussed with Jeremy and Jeff and it might be useful to read it.

Reply:
We can always comment this code out, but I think the times involved are quite small compared to the other time for other steps. 

As mentioned in a previous post, it is much easier to have the option to work with just the curation_id and not have to retrieve the entire SpikeSorting key to get the curation one wants to work with.

Since we have the curation num, I think people could easily use that if they want to, but I definitely want is to have the option to have a simple key that lets a user pick out a given curation quickly and easily.

Reply:
If it is `much easier to have the option to work with just the curation_id` then why not also have just a `sorting_id` for all the sortings, `recoridng_id` for all the recordings, `workspace_id` for all workspaces, `session_id` for all sessions etc? the user is already keeping a large dict as key from having worked with `SpikeSorting` etc upstream. I thought we had committed to a particular philosophy in designing the database (and much ink was spilled to justify it) and I'm confused about why we're backtracking.

going back to Jeremy's point from earlier, how would the user find out about this simple (but completely random) key associated with that sorting? Let's say I enter an entry to this table with this function. it returns a curation id. I lose it. then how do I look it up? 

anyway if this is what we're doing I suggest moving the foreign key from SpikeSorting to non-primary key